### PR TITLE
Strong mobile passwords

### PR DESCRIPTION
**Pull request by emord**
First part of https://docs.google.com/document/d/1Ta9TMnDHxNbYT-oa8_3HbejyK0wsSJXkXD_MbMHZIZY/edit#

This enables a project setting to force strong mobile passwords and uses the same password check as web users on those settings.

@calellowitz may have the most context on this.

:blowfish: / :blowfish: might be easiest

buddy @NoahCarnahan

### CODE REVIEW

File: corehq/apps/users/tests/bulk_upload.py [hunk 1]
```diff
@@ -86,8 +86,7 @@ def test_location_lookup_caching(self):
             self.assertEqual(get_supply_point.call_count, 1)
 
 
-class TestUserBulkUpload(TestCase, DomainSubscriptionMixin):
-
+class TestUserBulkUpload(TestCase, DomainSubscriptionMixin):
```
Comment (dimagimon) [inline corehq/apps/users/tests/bulk_upload.py]: W291 trailing whitespace

File: corehq/apps/users/tests/forms.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    def test_weak_password(self):
+        form = self.form("Taylor")
+        self.assertFalse(form.is_valid())
+
+    def test_strong_password(self):
+        form = self.form("TaylorSwift89!")
+        self.assertTrue(form.is_valid())
+
+class TestWeakSetUserPasswordForm(TestCase):
```
Comment (dimagimon) [inline corehq/apps/users/tests/forms.py]: E302 expected 2 blank lines, found 1

Comment (dimagimon) [inline corehq/apps/users/tests/forms.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/style/static/style/ko/knockout_bindings.ko.js [hunk 3]
```diff
@@ -687,3 +687,13 @@ ko.bindingHandlers.popover = {
...
 };
+
+ko.bindingHandlers.initializeValue = {
+    init: function(element, valueAccessor) {
+        valueAccessor()(element.getAttribute('value'));
+    },
+    update: function(element, valueAccessor) {
+        var value = valueAccessor();
+        element.setAttribute('value', ko.utils.unwrapObservable(value));
+    }
```
Comment (dimagimon) [inline corehq/apps/style/static/style/ko/knockout_bindings.ko.js]: Missing trailing comma. (comma-dangle)

**Comment by emord** [comment]
Added some commits for generating new passwords automatically. new buddy @czue

**Comment by czue** [comment]
@emord let me know if you need me to do buddy duties, but otherwise will defer to @calellowitz on this

File: corehq/apps/users/forms.py [hunk 4]
```diff
@@ -371,6 +371,7 @@ def __init__(self, project, user_id, **kwargs):
     def clean_new_password1(self):
         if self.project.strong_mobile_passwords:
             return clean_password(self.cleaned_data.get('new_password1'))
+        return self.clean_data.get('new_password1')
```
Comment (calellowitz) [inline corehq/apps/users/forms.py]: should this be `cleaned_data`?

Comment (emord) [inline corehq/apps/users/forms.py]: It was fixed in a later commit https://github.com/dimagi/commcare-hq/pull/12615/commits/6780ce54d5a6abf70c3dff651ce5cf4b244e431b

File: corehq/apps/style/static/style/ko/knockout_bindings.ko.js (hunk 3 above)
Comment (calellowitz) [inline corehq/apps/style/static/style/ko/knockout_bindings.ko.js]: why was this necessary? it looks fine but it seems like it would affect every knockout model we have, just trying to figure out what it changes

Comment (calellowitz) [inline corehq/apps/style/static/style/ko/knockout_bindings.ko.js]: ignore this comment, i see how you use it

**Comment by calellowitz** [comment]
@emord looks good overall. when are passwords generated server side vs client side? I see the code for both, but both seem to happen when `strong_mobile_passwords` is set. Otherwise, looks good to merge

**Comment by emord** [comment]
@calellowitz Should have mentioned this in the initial PR. 

Generating passwords for new workers is done in js, because you can create many mobile workers without refreshing, so js was easier.

Generating passwords for existing workers is done in python. The form is created in crispy forms so it was easiest to add an initial value there. That page is controlled by a bit of a hodgepodge of vanilla js and knockout and thought cleaning that up was a bit out of scope

**Comment by emord** [comment]
:white_check_mark:
